numba/core/types/containers.py

```diff
@@ -642,6 +673,45 @@ def unify(self, typingctx, other):
             if not other.is_precise():
                 return self
 
+    @property
+    def key(self):
+        return self.key_type, self.value_type, str(self.initial_value)
+
+
+class LiteralStrKeyDict(Literal, NamedTuple):
+    def __init__(self, literal_value, value_index=None):
+        self._literal_init(literal_value)
+        self.value_index = value_index
+        from collections import namedtuple
+        strkeys = [x.literal_value for x in literal_value.keys()]
+        self.tuple_ty = namedtuple('_ntclazz', ' '.join(strkeys))
+        self.tuple_inst = self.tuple_ty(*literal_value.values())
+        tys = [unliteral(x) for x in literal_value.values()]
+        NamedTuple.__init__(self, tys, self.tuple_ty)
+        self.name = 'LiteralStrKey[Dict]({})'.format(literal_value)
+        literal_vals = [getattr(x, 'literal_value', None) for x in literal_value.values()]
+
+    def __unliteral__(self):
+        return self
+
+    def unify(self, typingctx, other):
+        """
+        Unify this with the *other* one.
+        """
+        if isinstance(other, LiteralStrKeyDict):
+            for (k1, v1), (k2, v2) in zip(self.literal_value.items(),
+                                          other.literal_value.items()):
+                if k1 != k2: # keys must be same
+                    break
```

Comment:
Branch not covered.

Reply:
likely, unify is a bit of a mess because the `__unliteral__`s are not behaving.

Reply:
Think this is fixed.